staging/src/k8s.io/cli-runtime/pkg/resource/kustomizevisitor.go

```diff
@@ -38,7 +38,7 @@ type KustomizeVisitor struct {
 // Visit passes the result of a kustomize build to a StreamVisitor.
 func (v *KustomizeVisitor) Visit(fn VisitorFunc) error {
 	kOpts := krusty.MakeDefaultOptions()
-	kOpts.DoLegacyResourceSort = true
+	kOpts.Reorder = krusty.ReorderOptionLegacy
```

Comment:
can we confirm this gives equivalent behavior so kubectl maintains compatibility?

Reply:
Yes, the kustomize repo has a test for this: https://github.com/kubernetes-sigs/kustomize/blob/ce3e394a414387ce09679523902e981414b09a1a/api/krusty/legacy_order_test.go#L42